### ISSUE TITLE
Ubuntu 14.04: Fail to start prometheus container

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla-monitoring#415]
This problem doesn't exist with CentOS7 Ubuntu 16.04
It occurred 100% with Ubuntu 14.04.

```
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ cd /var/lib/scylla/scylla-grafana-monitoring-branch-2.0
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$  sudo service docker start || true
start: Job is already running: docker
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ mkdir -p /var/lib/scylla/scylla-grafana-monitoring-data
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ sudo chmod 777 /var/lib/scylla/scylla-grafana-monitoring-data

scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ ls -l /var/lib/scylla/
total 108
-rw-r--r-- 1 root root 101992 Oct 22 05:07 branch-2.0.zip
drwxr-xr-x 5 root root   4096 Oct 22 05:07 scylla-grafana-monitoring-branch-2.0
drwxrwxrwx 2 root root   4096 Oct 22 05:07 scylla-grafana-monitoring-data

```
```
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ sudo ./start-all.sh -s /var/lib/scylla/scylla-grafana-monitoring-branch-2.0/config/scylla_servers.yml -n /var/lib/scylla/scylla-grafana-monitoring-branch-2.0/config/node_exporter_servers.yml -d /var/lib/scylla/scylla-grafana-monitoring-data -v 2.2
Wait for alert manager container to start.Loading prometheus data from /var/lib/scylla/scylla-grafana-monitoring-data
0635f559f28aa5dfa0a9765fa3e11df1c638a440c9fb2ffa7ebcee9f5d95aee6
Wait for Prometheus container to start........Error: Prometheus container failed to start
scylla-test@rolling-upgrade-u14-2-3-monitor-node-0709efe2-0-1:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ sudo docker logs aprom
level=info ts=2018-10-22T05:33:54.814535468Z caller=main.go:222 msg="Starting Prometheus" version="(version=2.3.2, branch=HEAD, revision=71af5e29e815795e9dd14742ee7725682fa14b7b)"
level=info ts=2018-10-22T05:33:54.81465433Z caller=main.go:223 build_context="(go=go1.10.3, user=root@5258e0bd9cc1, date=20180712-14:02:52)"
level=info ts=2018-10-22T05:33:54.814753948Z caller=main.go:224 host_details="(Linux 4.4.0-130-generic #156~14.04.1-Ubuntu SMP Thu Jun 14 13:51:47 UTC 2018 x86_64 0635f559f28a (none))"
level=info ts=2018-10-22T05:33:54.814792719Z caller=main.go:225 fd_limits="(soft=524288, hard=1048576)"
level=info ts=2018-10-22T05:33:54.816539089Z caller=main.go:533 msg="Starting TSDB ..."
level=info ts=2018-10-22T05:33:54.816873519Z caller=main.go:402 msg="Stopping scrape discovery manager..."
level=info ts=2018-10-22T05:33:54.816908411Z caller=main.go:416 msg="Stopping notify discovery manager..."
level=info ts=2018-10-22T05:33:54.816943241Z caller=main.go:438 msg="Stopping scrape manager..."
level=info ts=2018-10-22T05:33:54.817200164Z caller=main.go:412 msg="Notify discovery manager stopped"
level=info ts=2018-10-22T05:33:54.817247254Z caller=web.go:415 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2018-10-22T05:33:54.817765928Z caller=main.go:398 msg="Scrape discovery manager stopped"
level=info ts=2018-10-22T05:33:54.818479127Z caller=manager.go:464 component="rule manager" msg="Stopping rule manager..."
level=info ts=2018-10-22T05:33:54.818532069Z caller=manager.go:470 component="rule manager" msg="Rule manager stopped"
level=info ts=2018-10-22T05:33:54.818571984Z caller=notifier.go:512 component=notifier msg="Stopping notification manager..."
level=info ts=2018-10-22T05:33:54.818614093Z caller=main.go:587 msg="Notifier manager stopped"
level=info ts=2018-10-22T05:33:54.818650562Z caller=main.go:432 msg="Scrape manager stopped"
level=error ts=2018-10-22T05:33:54.818801027Z caller=main.go:596 err="Opening storage failed lock DB directory: open /prometheus/data/lock: read-only file system"


```

**Comment by trtg** [comment]
seconding this issue- I got a slightly different error message on ubuntu, despite trying to change ownership of /prometheus/data to ubuntu. Sticking to older version for now unless someone knows what's up.

**Comment by amoskong** [comment]
Thanks for your reply. @trtg 

Related:
https://groups.google.com/forum/#!topic/prometheus-users/4qiYbUAPoXc
https://github.com/coreos/prometheus-operator/issues/830

I tried to change ownership, but it doesn't work.
```
sudo chown -R nobody:docker scylla-grafana-monitoring-data/
```

**Comment by yarongilor** [comment]
reproduced in Debian-8 as well.
```
admin@ip-172-30-0-21:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ sudo ./start-all.sh -s /var/lib/scylla/scylla-grafana-monitoring-branch-2.0/config/scylla_servers.yml -n /var/lib/scylla/scylla-grafana-monitoring-branch-2.0/config/node_exporter_servers.yml -d /var/lib/scylla/scylla-grafana-monitoring-data -l -v 2.3
Wait for alert manager container to start.Loading prometheus data from /var/lib/scylla/scylla-grafana-monitoring-data
105e207eb1e13f59eb95cc51edf048362a55826af03032f20c6d7fa244a4602e
Wait for Prometheus container to start........Error: Prometheus container failed to start
admin@ip-172-30-0-21:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ sudo docker logs aprom
level=info ts=2018-12-20T10:37:45.448436444Z caller=main.go:222 msg="Starting Prometheus" version="(version=2.3.2, branch=HEAD, revision=71af5e29e815795e9dd14742ee7725682fa14b7b)"
level=info ts=2018-12-20T10:37:45.448532092Z caller=main.go:223 build_context="(go=go1.10.3, user=root@5258e0bd9cc1, date=20180712-14:02:52)"
level=info ts=2018-12-20T10:37:45.448607742Z caller=main.go:224 host_details="(Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1 (2016-12-30) x86_64 ip-172-30-0-21 (none))"
level=info ts=2018-12-20T10:37:45.448645342Z caller=main.go:225 fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2018-12-20T10:37:45.449897409Z caller=main.go:533 msg="Starting TSDB ..."
level=info ts=2018-12-20T10:37:45.45032986Z caller=main.go:402 msg="Stopping scrape discovery manager..."
level=info ts=2018-12-20T10:37:45.45037454Z caller=main.go:416 msg="Stopping notify discovery manager..."
level=info ts=2018-12-20T10:37:45.450408729Z caller=main.go:438 msg="Stopping scrape manager..."
level=info ts=2018-12-20T10:37:45.45044135Z caller=main.go:412 msg="Notify discovery manager stopped"
level=info ts=2018-12-20T10:37:45.450482313Z caller=web.go:415 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2018-12-20T10:37:45.451142281Z caller=main.go:398 msg="Scrape discovery manager stopped"
level=info ts=2018-12-20T10:37:45.451694659Z caller=manager.go:464 component="rule manager" msg="Stopping rule manager..."
level=info ts=2018-12-20T10:37:45.451738867Z caller=manager.go:470 component="rule manager" msg="Rule manager stopped"
level=info ts=2018-12-20T10:37:45.451781714Z caller=notifier.go:512 component=notifier msg="Stopping notification manager..."
level=info ts=2018-12-20T10:37:45.451819192Z caller=main.go:587 msg="Notifier manager stopped"
level=info ts=2018-12-20T10:37:45.451856293Z caller=main.go:432 msg="Scrape manager stopped"
level=error ts=2018-12-20T10:37:45.451975781Z caller=main.go:596 err="Opening storage failed lock DB directory: open /prometheus/data/lock: read-only file system"
admin@ip-172-30-0-21:/var/lib/scylla/scylla-grafana-monitoring-branch-2.0$ cat /etc/debian_version 
8.7
```

**Comment by amnonh** [comment]
See #488 
@amoskong can you verify that the issue is closed (it's already in manager 2.2)

**Comment by amnonh** [comment]
@amoskong can we close this?
